web-registry/src/components/atoms/WithLoader.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { CircularProgress } from '@mui/material';
+
+type Props = {
+  isLoading: boolean;
+  children: JSX.Element;
+};
+
+const WithLoader = ({ isLoading, children }: Props): JSX.Element => {
+  if (isLoading) {
+    return <CircularProgress color="secondary" />;
+  }
+
+  return children;
+};
+
+export default WithLoader;
```

Comment:
This may function a little differently, but we have a similar component already, `regen-web/web-components/src/components/loading/Loading.tsx` FYI

Reply:
Yes, I saw this one but it had styling inside that was not necessary for where I wanted to use the loader.